### PR TITLE
Add validation for the TopN ops/kernels field

### PR DESCRIPTION
**Pull request by TomWildenhain-Microsoft**

### CODE REVIEW

File: tb_plugin/fe/src/utils/top.ts [hunk 1]
```diff
@@ -30,8 +30,13 @@ export function useTopN(options?: IOptions) {
...
+    setActualDebounce(useTop === UseTop.Use && topIsValid(topText) ? Number(topText): actualTop)
+  }, [topText, useTop])
 
-  return [top, actualTop, useTop, setTop, setUseTop] as const
+  return [topText, actualTop, useTop, setTopText, setUseTop] as const
+}
+
+export function topIsValid(topText: string) {
+  const top = Number(topText)
+  return top > 0 && Number.isInteger(top)
```
Comment (guotuofeng) [inline tb_plugin/fe/src/utils/top.ts:41]: need we us isNaN for top?

File: tb_plugin/fe/src/utils/top.ts [hunk 2]
```diff
@@ -30,8 +30,13 @@ export function useTopN(options?: IOptions) {
     ? React.useCallback(debounce(setActualTop, options.wait ?? 500), [])
     : setActualTop
   React.useEffect(() => {
-    setActualDebounce(useTop === UseTop.Use && top > 0 ? top : undefined)
-  }, [top, useTop])
+    setActualDebounce(useTop === UseTop.Use && topIsValid(topText) ? Number(topText): actualTop)
```
Comment (guotuofeng) [inline tb_plugin/fe/src/utils/top.ts:33]: in case of not valid, what the actualtop will be?

**Review by guotuofeng — CHANGES_REQUESTED** [review]

File: tb_plugin/fe/src/utils/top.ts (hunk 1 above)
Comment (TomWildenhain-Microsoft) [reply tb_plugin/fe/src/utils/top.ts:41]: For style I probably should. Technically NaN > 0 is false

File: tb_plugin/fe/src/utils/top.ts (hunk 2 above)
Comment (TomWildenhain-Microsoft) [reply tb_plugin/fe/src/utils/top.ts:33]: The previous valid value.  If the user enters something invalid, the graph will not be updated until a valid value is provided.

**Pull request closed without merging**